jetty-ee11/jetty-ee11-webapp/src/test/java/org/acme/webapp/ClassLoaderGetResourcesServlet.java

```diff
@@ -0,0 +1,46 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.acme.webapp;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class ClassLoaderGetResourcesServlet extends HttpServlet
+{
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
```

Comment:
I wonder if it is possible to change this servlet such as it also does `ServiceLoader.load(HttpFieldPreEncoder.class)` as this is closer to the original issue which states _If a webapp uses the ServiceLoader..._?

This is tricky as the `HttpFieldPreEncoder` is not visible in this servlet's classloader so that might not be possible.

Reply:
I'll see if I can find something that fits the ServiceLoader scenario too. (a different test)